src/com/ibm/security/appscan/altoromutual/servlet/LoginServlet.java

```diff
@@ -0,0 +1,107 @@
+vulnerabilities that were created expressly for demonstrating the functionality of
+application security testing tools. These vulnerabilities may present risks to the
+technical environment in which the application is installed. You must delete and
+uninstall this demonstration application upon completion of the demonstration for
+which it is intended. 
+
+IBM DISCLAIMS ALL LIABILITY OF ANY KIND RESULTING FROM YOUR USE OF THE APPLICATION
+OR YOUR FAILURE TO DELETE THE APPLICATION FROM YOUR ENVIRONMENT UPON COMPLETION OF
+A DEMONSTRATION. IT IS YOUR RESPONSIBILITY TO DETERMINE IF THE PROGRAM IS APPROPRIATE
+OR SAFE FOR YOUR TECHNICAL ENVIRONMENT. NEVER INSTALL THE APPLICATION IN A PRODUCTION
+ENVIRONMENT. YOU ACKNOWLEDGE AND ACCEPT ALL RISKS ASSOCIATED WITH THE USE OF THE APPLICATION.
+
+IBM AltoroJ
+(c) Copyright IBM Corp. 2008, 2013 All Rights Reserved.
+ */
+package com.ibm.security.appscan.altoromutual.servlet;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import com.ibm.security.appscan.Log4AltoroJ;
+import com.ibm.security.appscan.altoromutual.util.DBUtil;
+import com.ibm.security.appscan.altoromutual.util.ServletUtil;
+
+/**
+ * This servlet processes user's login and logout operations
+ * Servlet implementation class LoginServlet
+ * @author Alexei
+ */
+public class LoginServlet extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+	
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+    public LoginServlet() {
+        super();
+    }
+
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		//log out
+		try {
+			HttpSession session = request.getSession(false);
+			session.removeAttribute(ServletUtil.SESSION_ATTR_USER);
+		} catch (Exception e){
+			// do nothing
+		} finally {
+			response.sendRedirect("index.jsp");
+		}
+		
+	}
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		//log in
+		// Create session if there isn't one:
+		HttpSession session = request.getSession(true);
+
+		String username = null;
+		
+		try {
+			username = request.getParameter("uid");
+			if (username != null)
+				username = username.trim().toLowerCase();
+			
+			String password = request.getParameter("passw");
+			password = password.trim().toLowerCase(); //in real life the password usually is case sensitive and this cast would not be done
+			
+			if (!DBUtil.isValidUser(username, password)){
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; SQL Injection

Unsanitized input from `an HTTP parameter` flows into `executeQuery`, where it is used in an SQL query. This may result in an SQL Injection vulnerability.

<h6>
Line 82 | <a href="https://cwe.mitre.org/data/definitions/89.html">CWE-89</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 821</a> | <a href="https://learn.snyk.io/lesson/sql-injection/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 15 steps</b></summary>

`Step 1 - 3` https://github.com/snykyzach/zach-goof/blob/835de12359cb11ea30851c9741ef9582cd6e961a/src/com/ibm/security/appscan/altoromutual/servlet/LoginServlet.java#L75

`Step 4 - 7` [src/com/ibm/security/appscan/altoromutual/servlet/LoginServlet.java#L77](https://github.com/snykyzach/zach-goof/blob/835de12359cb11ea30851c9741ef9582cd6e961a/src/com/ibm/security/appscan/altoromutual/servlet/LoginServlet.java#L77)

`Step 8` [src/com/ibm/security/appscan/altoromutual/servlet/LoginServlet.java#L82](https://github.com/snykyzach/zach-goof/blob/835de12359cb11ea30851c9741ef9582cd6e961a/src/com/ibm/security/appscan/altoromutual/servlet/LoginServlet.java#L82)

`Step 9` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L212](https://github.com/snykyzach/zach-goof/blob/835de12359cb11ea30851c9741ef9582cd6e961a/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L212)

`Step 10` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L213](https://github.com/snykyzach/zach-goof/blob/835de12359cb11ea30851c9741ef9582cd6e961a/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L213)

`Step 11 - 15` https://github.com/snykyzach/zach-goof/blob/835de12359cb11ea30851c9741ef9582cd6e961a/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L219

</details>